x-pack/platform/plugins/shared/task_manager/server/lib/get_next_run_at.ts

```diff
@@ -15,11 +16,23 @@ export function getNextRunAt(
   const taskDelay = startedAt!.getTime() - runAt.getTime();
   const scheduleFromDate = taskDelay < taskDelayThresholdForPreciseScheduling ? runAt : startedAt!;
 
-  // Ensure we also don't schedule in the past by performing the Math.max with Date.now()
-  const nextCalculatedRunAt = Math.max(
-    intervalFromDate(scheduleFromDate, schedule!.interval)!.getTime(),
-    Date.now()
-  );
+  const { rrule, interval } = schedule || {};
+
+  let nextRunAt: Date | undefined | null;
 
+  if (interval) {
+    nextRunAt = intervalFromDate(scheduleFromDate, interval);
+  } else if (rrule) {
+    const _rrule = new RRule({
+      ...rrule,
+      dtstart: scheduleFromDate,
+    });
+    // adding 1ms to ensure the next run is always in the future
+    // if scheduleFromDate is equal to now (very low possibility), the next run will be now again, which causes loops
+    nextRunAt = _rrule.after(new Date(Date.now() + 1));
```

Comment:
can this throw an error ever? should we be catching anything?

Reply:
It is supposed to not throw with our limited configs, but still a possibility. 
In this case it would be caught in the `processResult` which sets the result as `Failed` and logs the error.
Idk what else we can do if we catch the error here. Maybe an extra log and throw again?

Reply:
In the usage in `get_first_run`, we have `rrule.after(now)?.toISOString() || nowString;`. Does that mean `.after` can return undefined? Do we need a fallback here?

Reply:
done, wrapped with try-catch and aded a test